src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenUpdateCaseStateAat.java

```diff
@@ -43,8 +43,8 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
 
         CaseData data = details.getData();
 
-        State state = State.valueOf(data.getApplicant2().getSolicitor().getAddress());
-        data.getApplicant2().getSolicitor().setAddress(null);
+        State state = State.valueOf(data.getApplicant2().getMiddleName());
+        data.getApplicant2().setMiddleName(null);
```

Comment:
for unit test you can just set ` caseDetails.setState(State.Holding);` instead of using data in the case.
Currently it is looking like you are setting middle state from middle name

Reply:
In your test you can do 
```
 caseDetails.setState(State.Holding);
        final AboutToStartOrSubmitResponse<CaseData, State> response = citizenUpdateCaseStateAat.aboutToSubmit(caseDetails, caseDetails);

```

Reply:
@nitinprabhu the input from that middle name field should be what's setting the state of the case. As that's how we'll update the state from one request

Reply:
so that's why the unit test is testing that the state changes if the middle name field is set

Reply:
ok got you